### ISSUE TITLE
Can't update my email address in the profile settings

### ISSUE THREAD

**Issue by MidouWebDev** [divanov11/Mumble#374]
### Preflight Checklist

* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

#

### Describe the bug

I can't update my email address in the profile settings

#

### To Reproduce

1. Go to 'mumble.dev'
2. Click on 'profile'
3. Scroll down to 'settings'
4. Try to update the email address
5. See Error

#

### Expected behavior

I expected to be able to update my email address like for the avatar

#

**Comment by Leopold-V** [comment]
Hi @MidouWebDev
I think it is a backend issue because whatever we modify in the user setting page, a request is sent to the api/users/profile_update/ route and always answer with an undefined email field  : https://i.gyazo.com/9f3172e087236a062e49a0c17cac74bc.png .

**Comment by MidouWebDev** [comment]
@Leopold-V thanks bro

I will check this with @PraveenMalethia

**Comment by MidouWebDev** [comment]
@PraveenMalethia

**Comment by ishitb** [comment]
> Hi @MidouWebDev
> I think it is a backend issue because whatever we modify in the user setting page, a request is sent to the api/users/profile_update/ route and always answer with an undefined email field : https://i.gyazo.com/9f3172e087236a062e49a0c17cac74bc.png .

An addition to this. Looking at the code it seems like the issue is that the 'api/users/profile_update/' endpoint, only updates the fields that are in the [UserProfile](https://github.com/divanov11/mumbleapi/blob/ef1399223d406419296633d13a004a77baef41ca/users/models.py#L25) model, and since the actual email is part of the django user model, it's not being updated.

**Comment by MidouWebDev** [comment]
> > Hi @MidouWebDev
> > I think it is a backend issue because whatever we modify in the user setting page, a request is sent to the api/users/profile_update/ route and always answer with an undefined email field : https://i.gyazo.com/9f3172e087236a062e49a0c17cac74bc.png .
> 
> An addition to this. Looking at the code it seems like the issue is that the 'api/users/profile_update/' endpoint, only updates the fields that are in the [UserProfile](https://github.com/divanov11/mumbleapi/blob/ef1399223d406419296633d13a004a77baef41ca/users/models.py#L25) model, and since the actual email is part of the django user model, it's not being updated. 

Yeah that's what @Leopold-V mentionned, i am waiting for @PraveenMalethia '@s opinion

**Comment by MidouWebDev** [comment]
#432

**Comment by MidouWebDev** [comment]
@PraveenMalethia we need an endpoint here

**Comment by PraveenMalethia** [comment]
Oh, I see ,  will look at that !

**Comment by PraveenMalethia** [comment]
i have updated the view to also modify the email field, which is of the default user model provided by Django !

**Comment by mani-barathi** [comment]
@MidouWebDev I'll work on this!

**Comment by MidouWebDev** [comment]
@mani-barathi okey, thanks !

**Comment by mani-barathi** [comment]
@MidouWebDev There is nothing to change in the frontend regarding the email update.
If the user tries to update the email, we are making a proper request to the backend with the new email as a payload. But in the response, the email field is not present. 
I checked the backend route I think line number **248** should be after line number **256**, such that response is created after the user object being updated with the new email.

![image](https://user-images.githubusercontent.com/49336839/120913822-adac5800-c6b7-11eb-885f-1d2a567232c9.png)

@PraveenMalethia what's your thought on this?

**Comment by MidouWebDev** [comment]
@PraveenMalethia

**Comment by PraveenMalethia** [comment]
check the `new_email` which is received in the 'email' payload and if that is None! then it will be assigned to the user profile!

**Comment by mani-barathi** [comment]
@PraveenMalethia Email is not received in the `user` or the `profile` object.

**Comment by okumujustine** [comment]
Looks like this is already working on my side, can someone close it!? or it's still being worked on? @PraveenMalethia

**Comment by okumujustine** [comment]
However, the frontend changes reflect after refreshing the page.
![ezgif com-gif-maker (2)](https://user-images.githubusercontent.com/22288040/122689181-74ccc100-d229-11eb-9fdf-b70ba52d0a44.gif)

**Comment by mani-barathi** [comment]
I get that. The backend is updating the email but it's not sending the updated email back in the response. Because of that we are not able to update the email with new email in front-end.

**Comment by okumujustine** [comment]
> I get that. The backend is updating the email but it's not sending the updated email back in the response. Because of that we are not able to update the email with new email in front-end.

Okay, i get that @mani-barathi .

A temporary solution would be to monitor success (**in .then**) where the API  call is being made from and set the email to the entered email. 
Or set to the previous email in case of API call failure (**in .catch**)
if backend will delay.

**Comment by mani-barathi** [comment]
@MidouWebDev should I implement the temporary fix suggested by @okumujustine

**Comment by okumujustine** [comment]
> @MidouWebDev should I implement the temporary fix suggested by @okumujustine

That will be great, @Mr-spShuvo you could assign @mani-barathi  on this if it's okay.

**Comment by PraveenMalethia** [comment]
Backend is updated to return updated email in response. Now you can update the frontend!

**Comment by okumujustine** [comment]
@mani-barathi , you are work on this !?

**Comment by mani-barathi** [comment]
@okumujustine I did make a simple temporary change, but if you what to take this issue go ahead and do it. 
Here is the link to [PR](https://github.com/divanov11/Mumble/pull/461) I made. Mr-spShuvo was requesting some changes so have a look into that.
